x/lockup/keeper/lock.go

```diff
@@ -499,6 +535,46 @@ func (k Keeper) ForceUnlock(ctx sdk.Context, lock types.PeriodLock) error {
 	return k.unlockInternalLogic(ctx, lock)
 }
 
+func (k Keeper) BreakLockForUnpool(ctx sdk.Context, lock types.PeriodLock) error {
+	return k.breakLockInternalLogic(ctx, lock)
+}
+
+func (k Keeper) breakLockInternalLogic(ctx sdk.Context, lock types.PeriodLock) error {
+	owner, err := sdk.AccAddressFromBech32(lock.Owner)
+	if err != nil {
+		return err
+	}
+
+	// send coins back to owner
+	if err := k.bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, owner, lock.Coins); err != nil {
+		return err
+	}
+
+	// change lock status to unlocking if it lock wasn't unlocking
+	if !lock.IsUnlocking() {
+		err = k.beginForceUnlock(ctx, lock, lock.Coins)
+		if err != nil {
+			return err
+		}
+	}
+
+	k.deleteLock(ctx, lock.ID)
+
+	// delete lock refs from the unlocking queue
+	err = k.deleteLockRefs(ctx, types.KeyPrefixUnlocking, lock)
```

Comment:
```suggestion
        if lock.IsUnlocking() {
	        err = k.deleteLockRefs(ctx, types.KeyPrefixUnlocking, lock)
	} else {
	        err = k.deleteLockRefs(ctx, types.KeyPrefixNotUnlocking, lock)
	}
```

Reply:
I suspect this is causing NotUnlocking locks to not be deleted from lock refs, and making the lock iterable through reference, but the actual lock was deleted and causing panic.